src/main/java/com/uber/cadence/internal/replay/ActivityDecisionContext.java

```diff
@@ -137,6 +137,11 @@ void handleActivityTaskCompleted(HistoryEvent event) {
         byte[] result = attributes.getResult();
         BiConsumer<byte[], Exception> completionHandle = scheduled.getCompletionCallback();
         completionHandle.accept(result, null);
+      } else {
+        throw new Error(
```

Comment:
This is expected to be thrown on non deterministic change. Do you think it is more useful for the end user than the generic determinism error? Could we set this one as a cause to the generic one?

Reply:
I'll change it to throw NonDeterminisicWorkflowError.